msdk-io/msdk-io-mzml2/src/test/java/io/github/msdk/io/mzml2/MzMLFileWriterTest.java

```diff
@@ -0,0 +1,175 @@
+    intensityBuffer = scan5.getIntensityValues();
+    Assert.assertEquals(837, (int) scan5.getNumberOfDataPoints());
+    Float scan5maxInt =
+        MsSpectrumUtil.getMaxIntensity(intensityBuffer, scan5.getNumberOfDataPoints());
+    Assert.assertEquals(8.6E3f, scan5maxInt, 1E2f);
+
+
+    // Cleanup
+    rawFile.dispose();
+    newMzMLFile.dispose();
+    tempFile.delete();
+
+  }
+
+  @Test
+  public void testSRM() throws MSDKException, IOException {
+
+    // Import the file
+    String file = "SRM.mzML";
+    File inputFile = getResourcePath(file).toFile();
+    Assert.assertTrue(inputFile.canRead());
+    MzMLFileParser parser = new MzMLFileParser(inputFile);
+    RawDataFile rawFile = parser.execute();
+    Assert.assertNotNull(rawFile);
+    Assert.assertEquals(1.0, parser.getFinishedPercentage(), 0.0001);
+
+    // Export the file to a new mzML
+    File tempFile = File.createTempFile("msdk", ".mzML");
+    MzMLFileWriter exporter = new MzMLFileWriter(rawFile, tempFile,
+        MzMLCompressionType.NUMPRESS_LINPRED_ZLIB, MzMLCompressionType.ZLIB);
+    exporter.execute();
+    Assert.assertEquals(1.0, exporter.getFinishedPercentage(), 0.0001);
+
+    // Import the new mzML
+    parser = new MzMLFileParser(tempFile);
+    RawDataFile newMzMLFile = parser.execute();
+    Assert.assertNotNull(newMzMLFile);
+    Assert.assertEquals(1.0, parser.getFinishedPercentage(), 0.0001);
+
+    // The file has 37 chromatograms
+    List<Chromatogram> chromatograms = rawFile.getChromatograms();
+    Assert.assertNotNull(chromatograms);
+    Assert.assertEquals(37, chromatograms.size());
+
+    // 4th chromatogram
+    Chromatogram chromatogram = chromatograms.get(3);
+    Assert.assertEquals(new Integer(4), chromatogram.getChromatogramNumber());
+    Assert.assertEquals(ChromatogramType.MRM_SRM, chromatogram.getChromatogramType());
+    Assert.assertEquals(new Integer(1608), chromatogram.getNumberOfDataPoints());
+    Assert.assertEquals(new Integer(2), (Integer) chromatogram.getIsolations().size());
+    Assert.assertEquals(new Double(440.706), chromatogram.getIsolations().get(1).getPrecursorMz());
+    Assert.assertEquals(ActivationType.CID,
+        chromatogram.getIsolations().get(0).getActivationInfo().getActivationType());
+    Assert.assertEquals(0.01095, chromatogram.getRetentionTimes()[0], 0.0001);
+    Assert.assertEquals(38.500003814697266, chromatogram.getIntensityValues()[0], 0.0001);
+
+    // 1st chromatogram
+    chromatogram = chromatograms.get(0);
+    Assert.assertEquals(ChromatogramType.TIC, chromatogram.getChromatogramType());
+    Assert.assertEquals(0, chromatogram.getIsolations().size());
+
+    // Check m/z values
+    Assert.assertEquals(407.706, chromatograms.get(1).getMz(), 0.001);
+    Assert.assertEquals(1084.486, chromatograms.get(18).getMz(), 0.001);
+    Assert.assertEquals(1042.516, chromatograms.get(35).getMz(), 0.001);
+
+    rawFile.dispose();
+
+    // Cleanup
+    rawFile.dispose();
```

Comment:
Cleanup should be done either in a finally block or the file in the first place, or call `File.deleteOnExit()` to make sure that if jvm dies, the files are still deleted